sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -68,10 +91,14 @@ case class DropTable(
       // Users should be able to drop such kinds of tables regardless if there is an error.
       case e: Throwable => log.warn(s"${e.getMessage}", e)
     }
-    hiveContext.invalidateTable(tableName)
-    hiveContext.runSqlHive(s"DROP TABLE $ifExistsClause$tableName")
-    hiveContext.sessionState.catalog.dropTable(
-      TableIdentifier(tableName), ignoreIfNotExists = true)
+    hiveContext.invalidateTable(tableName.quotedString)
+    val ifExistsClause = if (ifExists) "IF EXISTS " else ""
+    if (isView) {
+      hiveContext.runSqlHive(s"DROP VIEW $ifExistsClause${tableName.quotedString}")
+    } else {
+      hiveContext.runSqlHive(s"DROP TABLE $ifExistsClause${tableName.quotedString}")
```

Comment:
I think we should remove these two `runSqlHive` calls. We just use SessionCatalog to handle metadata.